### PR TITLE
feat: improve extensibility of update password feature

### LINKED ISSUE #10443: Improve extensibility of update password feature
As part of #9630 we refactor the update password feature to come up with a cleaner implementation that is better extensible as well. This includes a number of changes:

- delegate controller logic to component service
- clean up component view logic, drop bootstrap for layout
- add inline code documentation
- leverage new lazy loaded service logic

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
closes #10443

### CODE REVIEW

File: feature-libs/user/profile/components/update-password/update-password.service.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+  );
+
+  update(): void {
+    if (!this.form.valid) {
+      this.form.markAllAsTouched();
+      return;
+    }
+
+    this.form.disable();
```
Comment (dunqan) [inline feature-libs/user/profile/components/update-password/update-password.service.ts]: Do we need this?

File: feature-libs/user/profile/components/update-password/update-password.service.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  protected onSuccess(): void {
+    this.globalMessageService.add(
+      { key: 'updatePasswordForm.passwordUpdateSuccess' },
+      GlobalMessageType.MSG_TYPE_CONFIRMATION
+    );
+    this.routingService.go({ cxRoute: 'home' });
+  }
+
+  protected resetForm(): void {
+    this.busy.next(false);
```
Comment (dunqan) [inline feature-libs/user/profile/components/update-password/update-password.service.ts]: Do we need this if we are setting it from isUpdating$ stream?

Comment (tobi-or-not-tobi) [reply feature-libs/user/profile/components/update-password/update-password.service.ts:72]: nope, left over.

File: feature-libs/user/profile/components/update-password/update-password.service.ts (hunk 1 above)
Comment (tobi-or-not-tobi) [reply feature-libs/user/profile/components/update-password/update-password.service.ts]: nope, left over. it's handled in the tap inside `  isUpdating$`

File: feature-libs/user/profile/components/update-password/update-password.service.ts (hunk 2 above)
Comment (tobi-or-not-tobi) [reply feature-libs/user/profile/components/update-password/update-password.service.ts]: I believe so, we're resetting busy to control the loading/updating state.

**Review by dunqan — APPROVED** [review]

**Pull request merged**